website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+        self.nameFilters = [];
+        // Calculate tag filters
+        self.tagFilters = [];
+
+        // Placeholder for node data
+        self.data = m.prop([]);
+
+        // Activity Logs
+        self.activityLogs = m.prop();
+        self.showMoreActivityLogs = m.prop(null);
+        self.getLogs = function _getLogs (nodeId, link, addToExistingList) {
+            var url = link || $osf.apiV2Url('nodes/' + nodeId + '/logs/', { query : { 'page[size]' : 6, 'embed' : ['nodes', 'user', 'linked_node', 'template_node']}});
+            var promise = m.request({method : 'GET', url : url, config : xhrconfig});
+            self.activityLogs([]); // Empty old logs first;
+            self.showMoreActivityLogs(null);
+            promise.then(function _getLogsThen (result){
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                    if(addToExistingList){
+                        self.activityLogs().push(log);
+                    }
+                });
+                if(!addToExistingList){
+                    self.activityLogs(result.data);  // Set activity log data
+                }
+                self.showMoreActivityLogs(result.links.next); // Set view for show more button
+            });
+            return promise;
+        };
+        // separate concerns, wrap getlogs here to get logs for the selected item
+        self.getCurrentLogs = function _getCurrentLogs ( ){
+            if(self.selected().length === 1){
+                var id = self.selected()[0].data.id;
+                var promise = self.getLogs(id);
+            }
+        };
+
+        /* filesData is the link that loads tree data. This function refreshes that information. */
+        self.updateFilesData = function _updateFilesData (linkObject) {
+            if (linkObject.link !== self.currentLink) {
+                self.updateBreadcrumbs(linkObject);
+                self.updateList(linkObject);
+                self.currentLink = linkObject.link;
+            }
+            self.showSidebar(false);
+        };
+
+        // INFORMATION PANEL
+        /* Defines the current selected item so appropriate information can be shown */
+        self.selected = m.prop([]);
+        self.updateSelected = function _updateSelected (selectedList){
+            self.selected(selectedList);
+            self.getCurrentLogs();
+        };
+
+        // USER FILTER
+        self.activeFilter = m.prop({});
+        self.updateFilter = function _updateFilter (filter) {
+            self.activeFilter(filter);
+            self.updateFilesData(filter);
+        };
+
+        self.removeProjectFromCollections = function _removeProjectFromCollection () {
+            // Removes selected items from collect
+            var collection = self.activeFilter().data.node;
+            self.selected().map(function _removeProjectFromCollectionsMap (item){
```

Comment:
@brianjgeiger Can you point me in the right direction about doing bulk request on this?